tests/validation/cattlevalidationtest/core/test_machine_amazon.py

```diff
@@ -0,0 +1,552 @@
+import boto.ec2
+import logging
+import traceback
+
+from common_fixtures import *  # NOQA
+
+DEFAULT_TIMEOUT = 450
+access_key = os.environ.get('AMAZON_ACCESSKEY')
+secret_key = os.environ.get('AMAZON_SECRETKEY')
+
+# Amazon settings used for instance deployment
+vpc_id = "vpc-eb2d928e"
+subnet_id = "subnet-5775c020"
+security_group = "test"
+zone = "a"
+region = "us-west-2"
+ami = "ami-23ebb513"
+size = "32"
+instance_type = "t1.micro"
+
+# Amazon settings used for instance deployment in default region
+vpc_id_in_east = "vpc-44144621"
+subnet_id_in_east = "subnet-22d1bd55"
+
+
+# Use amazon settings from environment variables , if set
+if os.environ.get('AMAZON_VPC_ID') is not None:
+    vpc_id = os.environ.get('AMAZON_VPC_ID')
+if os.environ.get('AMAZON_SUBNET_ID') is not None:
+    subnet_id = os.environ.get('AMAZON_SUBNET_ID')
+if os.environ.get('AMAZON_SECURITY_GROUP') is not None:
+    security_group = os.environ.get('AMAZON_SECURITY_GROUP')
+if os.environ.get('AMAZON_ZONE') is not None:
+    zone = os.environ.get('AMAZON_ZONE')
+if os.environ.get('AMAZON_REGION') is not None:
+    region = os.environ.get('AMAZON_REGION')
+if os.environ.get('AMAZON_AMI') is not None:
+    region = os.environ.get('AMAZON_AMI')
+if os.environ.get('AMAZON_VPC_ID_EAST') is not None:
+    vpc_id_in_east = os.environ.get('AMAZON_VPC_ID')
+if os.environ.get('AMAZON_SUBNET_ID_EAST') is not None:
+    subnet_id_in_east = os.environ.get('AMAZON_SUBNET_ID')
+
+
+# Default values set by Docker Machine
+default_image_in_west = "ami-898dd9b9"
+default_image_in_east = "ami-4ae27e22"
+default_security_group = "docker-machine"
+default_instance_type = "t2.micro"
+default_root_size = "16"
+default_zone = "a"
+default_region = "us-east-1"
+
+# Amazon error message validation
+
+error_msg_auth_failure = "code=401"
+error_msg_invalid_vpc = "unable to find a subnet in the zone"
+error_msg_invalid_subnet = "code=400"
+
+if_machine_amazon = pytest.mark.skipif(
+    os.environ.get('AMAZON_ACCESSKEY') is None or
+    os.environ.get('AMAZON_SECRETKEY') is None,
+    reason='Amazon EC2 Accesskey/Secretkey is not set')
+
+# Get logger
+ch = logging.StreamHandler()
+logger = logging.getLogger(__name__)
+logger.addHandler(ch)
+
+
+@pytest.fixture(scope='session', autouse=True)
+def register_host(admin_client):
```

Comment:
Since I am using this fixture with "autouse" to "true" and scope "session" , this method would be invoked once for every      test file . I did not want this method to be invoked repeatedly unless the test case needed them , so included it in the test file instead of common_fixtures.